File: app/src/main/java/uk/co/transferx/app/ui/splash/SplashActivity.java
package uk.co.transferx.app.ui.splash;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;

import javax.inject.Inject;

import uk.co.transferx.app.ui.base.BaseActivity;
import uk.co.transferx.app.R;
import uk.co.transferx.app.TransferXApplication;
import uk.co.transferx.app.ui.landing.LandingActivity;
import uk.co.transferx.app.ui.signin.SignInActivity;
import uk.co.transferx.app.ui.splash.presenter.SplashPresenter;
import uk.co.transferx.app.ui.welcom.WelcomeActivity;

/**
 * Created by sergey on 19.11.17.
 */

public class SplashActivity extends BaseActivity implements SplashPresenter.SplashUI {

    @Inject
    SplashPresenter presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((TransferXApplication) getApplication()).getAppComponent().inject(this);
        setContentView(R.layout.splash_activity_layout);

    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.attachUI(this);
    }


    @Override
    protected void onPause() {
        presenter.detachUI();
        super.onPause();
    }

    @Override
    public void goToWelcomeScreen() {
        // TEMPORARY LINK TO NEW LANDING ACTIVITY
        // TODO - REFACTOR LINK WHEN LINKING IS PERFORMED
        Intent intent = new Intent(this, LandingActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        finish();
    }

    @Override
    public void goToPinScreen() {
        SignInActivity.starSignInActivity(SplashActivity.this);
        finish();
    }

    @Override
    public void goToWelcome() {
        //no op
    }
}

File: app/src/main/java/uk/co/transferx/app/ui/splash/SplashActivity.kt
package uk.co.transferx.app.ui.splash

import android.content.Intent
import android.os.Bundle
import org.jetbrains.anko.*

import javax.inject.Inject

import uk.co.transferx.app.ui.base.BaseActivity
import uk.co.transferx.app.R
import uk.co.transferx.app.TransferXApplication
import uk.co.transferx.app.ui.landing.LandingActivity
import uk.co.transferx.app.ui.homescreen.MainActivity
import uk.co.transferx.app.ui.signin.SignInActivity
import uk.co.transferx.app.ui.signup.SignUpActivity
import uk.co.transferx.app.ui.splash.presenter.SplashPresenter
import uk.co.transferx.app.ui.tutorial.TutorialActivity
import uk.co.transferx.app.util.Constants
import uk.co.transferx.app.util.Constants.SIGNUP_PIN_STEP

/**
 * Created by sergey on 19.11.17.
 */

class SplashActivity : BaseActivity(), SplashContract.View {
    @Inject
    lateinit var presenter: SplashPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        (application as TransferXApplication).appComponent.inject(this)
    }

    override fun onResume() {
        super.onResume()
        presenter.attachUI(this)
    }

    override fun onPause() {
        super.onPause()
        presenter.detachUI()
    }

    override fun goToLandingScreen() {
        startActivity(intentFor<LandingActivity>().newTask().clearTask())
    }

    override fun goToSetPinScreen() {
        startActivity(intentFor<SignUpActivity>(SIGNUP_PIN_STEP to 2).newTask().clearTask())
    }

    override fun goToPinScreen() {
        // TODO - CREATE LINK TO PIN WHEN PIN SCREEN IS FINISHED AND LINKING IS PERFORMED
        startActivity(intentFor<MainActivity>().newTask().clearTask())

    }

    override fun goToTutorialScreen() {
        startActivity(intentFor<TutorialActivity>().newTask().clearTask())
    }

    // Method required by base UI. In further development, should be refined or deleted
    override fun goToWelcome() {
        //no op
    }
}
